### ISSUE TITLE
Automatic updates should be supplemented

### ISSUE THREAD

**Issue by uspBHV** [piwik/piwik#20301]
Matomo 4.13.3 asked me about my main concern in relation to Matomo. My answer: My main concern is that automatic updates do not complete and generate error messages that send me to the desert. My tip: Set up automatic updates so that the cache or, even better, the entire "tmp" folder is emptied at the same time. This can save a lot of time.

Varun Kothare (Web Support Technician), asked me if I would be able to create a feature request on the matomo issue tracker. So here I try: Automatic updates should be supplemented with the function of simultaneously emptying the "tmp" folder when running.

**Comment by sgiehl** [comment]
Hi @uspBHV 
Thanks for the suggestions. We actually do have a method that is responsible for clearing all related caches after an update.
See https://github.com/matomo-org/matomo/blob/151ebfd143b9c555a9e1e76157cb8371cf504d8a/core/Filesystem.php#L33-L56

Doesn't clearing the caches work correctly for you after updating?

**Comment by uspBHV** [comment]
I am not sure. The last two times I updated, Matomo returned a rather disconcerting error message. The first time I ignored the message hoping it would go away on its own. The second update with the same error message, I checked the "tmp" folder and then emptied it.

We have been using Matomo since 2008. I have never cleared a cache before and never had any problems with temporary files since 2008. I am now waiting for the upcoming update and hope that the error message mentioned at the beginning will not be repeated.

In the userforum I found a single post regarding an identical error message. However, the answers to it were confused and did not lead to any instructions for action.

Let's put it this way: The file "filesystem.php" does give instructions to delete all associated caches after automatic updates. But in my particular case this was not enough to clean the cache so thoroughly that no error could occur.

So, for the first time since 2008, an error message occurred. Emptying the "tmp" folder completely also has the perceived advantage that Matomo now works and displays more smoothly.

Emptying the folder "tmp" completely can also not hurt, because these are temporary files. I am generally not a friend of caches and turn them off where I can. If there is an inexplicable error message somewhere, it is usually the cache's fault - so I almost blindly clear everything that is temporary and can lead astray.

In short: Deleting temporary files completely is always the clean solution. Emptying the "tmp" folder completely is a proactive measure to avoid alienating error messages. That's all.

**Comment by sgiehl** [comment]
@uspBHV The cache clear mentioned above should actually remove all relevant files from the tmp folder. And as far as I know we did not change anything with clearing caches after update in the last releases, so it's a bit strange you only had the issue lately.

Did you maybe change anything in the meantime on your server? Like installing a new PHP version, maybe with other modules enabled. Or was your webserver maybe changed and there had been files in the tmp folder where there webserver did no longer have permission to remove them or similar?

**Comment by uspBHV** [comment]
Hallo Stefan, here is nothing to add to my previous, rather detailed report. No, to my knowledge there are no changes to the server configuration. If you, like me, can only guess, then we should end the discussion now. Thank you very much for your time. #

**Comment by sgiehl** [comment]
There is actually also a rather old issue around that topic here: #6402
I will close this one and put the other one in our prioritization queue, so we will check that topic again.
Thanks again for bringing this issue up.
